### PR TITLE
Add sailfish-developer layer

### PR DESCRIPTION
**Pull request by vityafx**
Adds `sailfish-developer` layer which provides hotkeys and spacemacs integration of the equally named emacs package.

### CODE REVIEW

**Comment by vityafx** [comment]
I don't quite understand the error. What it is?

```
!!!!!!!!!!!!!! TEST FAILED !!!!!!!!!!!!!!
current-buffer: README.org<2>
current-file: /root/.emacs.d/layers/+tools/sailfish-scratchbox/README.org
error: Link "img/logo.png" in file "/root/.emacs.d/layers/+tools/sailfish-scratchbox/README.org" has type "fuzzy" but the type isn't implemented in spacemacs//org-edn-link
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

error in process sentinel: (error Process worker was exited abnormally with code 1
)
Documentation needs some fixing ;)
Exited with code 1
```

**Comment by smile13241324** [comment]
@vityafx your documentation is not following certain conventions which is detected by the CI server. Review your image link in your documentation. You are not declaring a link type which is prohibited as it causes issues with various exports. It should be  like this `[[file:img/typewriter.jpg]]`.

Also your README.org does not conform to the document conventions as it does not contain a `features` block. Please review https://github.com/syl20bnr/spacemacs/blob/develop/core/templates/README.org.template for details. 

Also I have seen in your documentation you are still include headers which are only part of the template and should be replaced by your code, please have a look at the selectric or java layer for an example of how it should look like.

Also you are having a license issue in your package.el file. You wish to use "MIT" license but you can't as spacemacs is "GPLv3" and therefore all packages and layers do need to use that license too. Have a look at https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org#contributing-code.

**Comment by vityafx** [comment]
@smile13241324 thanks, I will fix everything next week.

**Comment by vityafx** [comment]
@smile13241324 In license part, did you mean that I can't have a GPLv3 layer which installs MIT packages?

**Comment by smile13241324** [comment]
@vityafx about the MIT package I am not sure but about the layer files the documentation is very clear. Those must be GPLv3.

@bmag do we have a problem with including MIT packages from MELPA?

**Comment by bmag** [comment]
There shouldn't be a problem to use MIT-licensed packages from Melpa. They aren't actually part of our code-base, and aren't part of the "layer files". However, the **`packages.el` file has to be GPL**, since it is a layer file.

File: layers/+tools/sailfish-scratchbox/packages.el [hunk 1]
```diff
@@ -0,0 +1,39 @@
+;;; packages.el --- sailfish scratchbox layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2017 Victor Polevoy
+;;
+;; Author: Victor Polevoy <fx@thefx.co>
+;; URL: https://github.com/vityafx/sailfish-scratchbox-layer
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: MIT
```
Comment (bmag) [inline layers/+tools/sailfish-scratchbox/packages.el]: Wrong license. This file needs to be GPLv3 to include it in Spacemacs.

File: layers/+tools/sailfish-scratchbox/packages.el [hunk 2]
```diff
@@ -0,0 +1,39 @@
+;;; packages.el --- sailfish scratchbox layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2017 Victor Polevoy
+;;
+;; Author: Victor Polevoy <fx@thefx.co>
+;; URL: https://github.com/vityafx/sailfish-scratchbox-layer
```
Comment (bmag) [inline layers/+tools/sailfish-scratchbox/packages.el]: URL should point be the URL of Spacemacs. Besides, the current URL gives me a 404 error - is it a private repo?

File: layers/+tools/sailfish-scratchbox/packages.el [hunk 3]
```diff
@@ -0,0 +1,39 @@
+;;; packages.el --- sailfish scratchbox layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2017 Victor Polevoy
```
Comment (bmag) [inline layers/+tools/sailfish-scratchbox/packages.el]: Copyright should be assigned to "Sylvain Benner & Contributors", as in other layers.

File: layers/+tools/sailfish-scratchbox/packages.el [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+;;
+;; Author: Victor Polevoy <fx@thefx.co>
+;; URL: https://github.com/vityafx/sailfish-scratchbox-layer
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: MIT
+;;; Commentary:
```
Comment (bmag) [inline layers/+tools/sailfish-scratchbox/packages.el]: Unnecessary - this file isn't a package itself. Please remove.

File: layers/+tools/sailfish-scratchbox/packages.el [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+;;
+;; Commands
```
Comment (bmag) [inline layers/+tools/sailfish-scratchbox/packages.el]: The entire "Commands" section is unnecessary - we've got README.org for that, and this file isn't a package. Please remove.

File: layers/+tools/sailfish-scratchbox/packages.el [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+;;
+;; SPC c s - sailfish scratchbox menu
+;; SPC c s b - sailfish scratchbox mb2 build
+;; SPC c s d - sailfish scratchbox deploy rpms
+;; SPC c s i - sailfish scratchbox install rpms
+
+;;; Code:
```
Comment (bmag) [inline layers/+tools/sailfish-scratchbox/packages.el]: "Code:" comment is unnecessary - this file isn't a package. Please remove.

File: layers/+tools/sailfish-scratchbox/packages.el [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      (spacemacs/declare-prefix "cs" "sailfish os scratchbox")
+      (spacemacs/set-leader-keys
+        "csb" 'sailfish-scratchbox-mb2-build
+        "csd" 'sailfish-scratchbox-deploy-rpms
+        "csi" 'sailfish-scratchbox-install-rpms
+      )
+    )
+  )
+)
+;;; packages.el ends here
```
Comment (bmag) [inline layers/+tools/sailfish-scratchbox/packages.el]: "ends here" comment is unnecessary - this file isn't ia package itself. Please remove.

File: layers/+tools/sailfish-scratchbox/packages.el [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    (progn
+      (spacemacs/declare-prefix "cs" "sailfish os scratchbox")
+      (spacemacs/set-leader-keys
+        "csb" 'sailfish-scratchbox-mb2-build
+        "csd" 'sailfish-scratchbox-deploy-rpms
+        "csi" 'sailfish-scratchbox-install-rpms
+      )
+    )
+  )
+)
```
Comment (bmag) [inline layers/+tools/sailfish-scratchbox/packages.el]: In accordance to Elisp conventions, should be:
```elisp
        "csi" 'sailfish-scratchbox-install-rpms))))
```

Not:
```elisp
        "csi" 'sailfish-scratchbox-install-rpms
      )
    )
  )
)
```

File: layers/+tools/sailfish-scratchbox/packages.el (hunk 2 above)
Comment (vityafx) [reply layers/+tools/sailfish-scratchbox/packages.el]: Ehm, it existed before I decided to move it into the spacemacs. I have deleted it already. Gonna change to URL of Spacemacs.

**Comment by vityafx** [comment]
@bmag I have changed everything you told.

**Comment by vityafx** [comment]
AFAIK I did everything here at dec 22, 2017. Should I do anything else, @bmag ?

**Comment by vityafx** [comment]
@bmag Could you help me please? I don't understand what I should do.

**Comment by sdwolfz** [comment]
@vityafx This looks good to me, I think you need to have patience until someone with merging powers decides it's the appropriate time to do so.

**Comment by vityafx** [comment]
@sdwolfz Thank you very much for your feedback! I thought I had to do something but I did not know what exactly. However, I don't understand why the `changes requested` flag was not reset automatically after my commits.

**Comment by vityafx** [comment]
@sdwolfz  Oh, thanks much! Again!

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs!
This PR has been cherry-picked into the `develop` branch. It's safe to delete your branch.
Commit: https://github.com/syl20bnr/spacemacs/commit/51b04beb268bf585c7f12f867060ba82d694b622

**Pull request closed without merging**